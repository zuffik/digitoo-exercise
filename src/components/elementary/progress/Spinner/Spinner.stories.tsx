import {Spinner} from './Spinner';

export default {
    title: 'Elementary/Progress/Spinner',
};

export const spinner = Spinner;
