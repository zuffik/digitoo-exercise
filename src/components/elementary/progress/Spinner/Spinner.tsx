import * as React from 'react';

interface Props {
  size?: number;
}

export const Spinner: React.FC<Props> = (props: Props) => {
  const size = props.size || 48;
  return <div className="spinner-border" style={{width: size + 'px', height: size + 'px'}} />;
};
