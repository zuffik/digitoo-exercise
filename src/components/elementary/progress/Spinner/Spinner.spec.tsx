import * as React from 'react';
import {render} from '@testing-library/react'
import {Spinner} from "./Spinner";

describe('<Spinner/>', () => {
    it('should render', () => {
        const {asFragment} = render(<Spinner/>);
        expect(asFragment()).toMatchSnapshot();
    });
});
