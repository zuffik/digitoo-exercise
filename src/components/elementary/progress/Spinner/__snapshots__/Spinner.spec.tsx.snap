// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner/> should render 1`] = `
<DocumentFragment>
  <div
    class="spinner-border"
    style="width: 48px; height: 48px;"
  />
</DocumentFragment>
`;
